crates/app-host/src/block_tracer.rs

```diff
@@ -0,0 +1,78 @@
+
+use anyhow::Result;
+
+use scroll_executor::BlockTrace;
+use tokio::runtime::Runtime;
+use std::sync::Arc;
+use l2_client::L2Client;
+
+pub struct BlockTracer {
+    l2_client: Arc<l2_client::L2Client>,
+    rt: Runtime,
+}
+
+impl BlockTracer {
+    pub fn new() -> Self {
+        todo!()
+    }
+
+    pub async fn get_block_traces(&self, blocks: Vec<u64>) -> Result<Vec<BlockTrace>> {
+        let mut handles = Vec::with_capacity(blocks.len());
+
+        for block in blocks {
+            let l2_client = self.l2_client.clone();
+            // todo: use a singleton tokio runtime with limited worker to spawn task
+            let handle = self.rt.spawn(async move {
+                BlockTracer::get_block_trace(l2_client, block).await
+            });
+            handles.push(handle);
+        }
+
+        let mut block_traces = Vec::with_capacity(handles.len());
```

Comment:
I think this could be simplified to something like `Result::Ok(tokio::join!(handles))`